CMakeLists.txt

```diff
@@ -858,6 +869,13 @@ if (Seastar_ALLOC_PAGE_SIZE)
     PUBLIC SEASTAR_OVERRIDE_ALLOCATOR_PAGE_SIZE=${Seastar_ALLOC_PAGE_SIZE})
 endif ()
 
+set_option_if_package_is_found (Seastar_COLORED_LOG fmt
+  VERSION 6.0.0)
+if (Seastar_COLORED_LOG)
+  target_compile_definitions (seastar
+    PRIVATE SEASTAR_COLORED_LOG)
+endif ()
+
```

Comment:
Sorry to only notice it now, but I think we can drop the whole cmake change and keep everything in log.cc, by looking at FMT_VERSION and maybe __has_include. It will end up much simpler, especially for people who hate cmake.

Reply:
what if user wants to explicitly enable/disable this feature instead of leaving the whole thing to the compile-time check?

Reply:
especially to those who don't want to enable this feature. without an option it'd be a little bit awkward to them to disable it -- they'd have to use `CPPFLAGS` or `CXXFLAGS` to define a macro as a knob for this purpose. this is not consistent when compared to other options and hence difficult to reason about.

Reply:
It's better to make it a run-time flag. Compile-time flags should be reserved for really critical things, nobody wants to recompile just to enable/disable log coloring.